src/Pyro4/util.py

```diff
@@ -369,7 +369,16 @@ def dict_to_class(cls, data):
                 exceptiontype = getattr(sqlite3, short_classname)
                 if issubclass(exceptiontype, BaseException):
                     return SerializerBase.make_exception(exceptiontype, data)
-
+        
+        # serpent serializes float('nan') as {'__class__':'float','value':'nan'}. 
+        #  I'm not sure if this is the most elegant way to handle that edge case 
+        #  (it'd likely make more sense for this knowledge to live in 
+        #  SerpentSerializer?) - but it does the trick. ~ jackjweinstein 1/26/17

```

Comment:
Yeah the better place is to put this special case in SerpentSerializer because the other serializers won't ever encode a float nan to this pattenr